examples/simple-hard-coded.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+// This example is used in the documentation.
+
+// 1. const { parseArgs } = require('node:util'); // from node
+// 2. const { parseArgs } = require('@pkgjs/parseargs'); // from package
+const { parseArgs } = require('..'); // in repo
+
+const args = ['-f', '--bar', 'b'];
+const options = {
+  foo: {
+    type: 'boolean',
+    short: 'f'
+  },
+  bar: {
+    type: 'string'
+  }
+};
+const {
+  values,
+  positionals
+} = parseArgs({ args, options });
```

Comment:
Thoughts on inlining `config.options` for all `/examples/*`?

Configuring `options` becomes a bit tedious when validating with TypeScript or JSDoc as inferred types are wide:
<img width="430" alt="image" src="https://user-images.githubusercontent.com/32409546/171065416-5842ac67-725d-4011-82d3-e9325f22f99a.png">

To resolve the errors authors are required to cast each `optionConfig.type` as `const` to narrow the types:
<img width="620" alt="image" src="https://user-images.githubusercontent.com/32409546/171065633-1a983b00-6625-4f96-8388-d31b5876abfb.png">

Example with `config.options` inlined:
<img width="449" alt="image" src="https://user-images.githubusercontent.com/32409546/171065694-aadf2d71-1957-4807-8463-ea2052be15fd.png">

Reply:
> To resolve the errors authors are required to cast each `optionConfig.type` as `const` to narrow the types:

Nah, you can just cast the whole thing:

```ts
const options = {
  foo: {
    type: 'boolean',
    short: 'f'
  },
  bar: {
    type: 'string'
  }
} as const;
```

Reply:
Yup, this works too. I'm aware of the various patterns for function argument inference and have a [playground](
https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCoAtQHNkG8BQyyA2gNYQCeAXMgM5hTYC6N9jIW+AvvvqJLEQoM2WnkLIYAeyk0RHANwSARnChzMi7rwRSQ9ZGE1ZaARg2jkAXnFFps20WSYaAckyuJXADQq1NAidnYDcPXyIfCQBXEFIQKQB3EAAFKCkAB2gwamR3YFdkAHpC5ABRKDSoZEBeDcBZHe1CgCp8AAEoCDAoqH1kXX0wQ2MzCw5afEbCnT0DI1EAJmtHewCJIhdcjy9w5FV1RzWQjfztyKIYuMSUtMyobND8opKAOSlkaEqawDkd7mQ4MT76Lwmq12p1uv9pgNZqM5isgu04AATPQAGwokhkcKCTgRyJAaOCNAARJgiUoglxyTiIEjUejdljschcXTCcgScAyatkJTuSz8ejzvEkqkMlkchyuTwJlN+oNRABmEYmRaBDEONUHe6eCLbBn7Nl5VwnbZCy6im53I4FYplCpSKp1HZRAbxZCgGDQCAgJBfH5-XqQoHNNodLo9AFQoZKtBDcaTfC2gDqmBRKGqgDEd34IBBdOCQAntdIOsBiUAANzgKOAiMMFEyYgSKBRUgMUi67pAnvaPogADoB0CSgBaUdj8cTydT6cz2fT2UGGvesDAbIAMRiCBXelMiwAPKg3gAPSAgRFieQmAB8AApoSY5ABKOTWK-y0YLqOiAAsiyX4FXCgNx9bcQFMG81WWA11iNE0-D2TUdjgAAvNxVGQ41uVtC1oAJd4HTEZQIAQOAoloFAAANTAo5BkQgMR4gGABbfMEHQQZKMPCAT29c9Y1EGjIygOB+CwkoKyrGtUHrCAaFMbxHjtD46i2CRbQACUSCBy2gBSzSSZB0muLJgHo2i9FcAZ8KqIiSLIlAjBQUB0hdOgGCiLc8xRAkWLANizMc5AKK4nizwvIZBMhYTRLOWJhSuMVbhyI1FJeN57Udb4uEfYMQTDcFAzle9aG-Rl1TKoJ1jYbAqQiWqkIQ7kgnQ1gGBq7leSCfSEstHJqq0aUE1tOcRtGsbRqHWMUFoKQURdYA9AHPtg2aZBGmQABJRFl0AyRN1AwZ8zoYAmPSQt8p6QKjKkcsl1rCjioohTlFcpFEVXBaQCrNFCHWytq0RfMAvQJyQBc5jWJBsRAuCiK60yY9EDAAk4DPTtuzEb7kCYiAmKIqBaGWtbJkjd1toA9d9s+hYbAPY9Tz40ojyRg8hgU1Ar1vYqn1fd8TE-PnaAAVj-cmV0pkDqYgiQoMQmCwlUoh9UQlrcnQzCgmw4yoDwjLCOI0jyKC6jzLMxjschjigpChnwoEwq2BE8AxM7f6pJkuSFNtcplPqXU1OeRJ0sqMQ9GQbrDO1lczOqQAZPeiOLzSj5KPEUn2HRqepssm8bc7z8bExKABVFdq2yeGzN0dJTNrGA0iYoLskyYdPXoCjC+cMAwHSWgqGKLBV3QKJlD7XQmMKWhQERdoZqgIfaEKJuIBb+iwEKZQW2UQpEXgVQAAYIGUOYhYVOA5kROYAE5TERb8hcvuZv0vvehYANm-BVlD30xP-fvfL7gHACe7YoBIEKNxJGfZER9lLAAYgADIKm-B3fOqC0GTl4NxYstwK7IFSCdD6OlrD4AAJAAB9kAgCiN5UhFCYjbRgKACAiJaFuXYJwchlCoh42gKw5QMg0yo1YbQCgeNZp8OAAPcASh8BYJLLggA0pQWgAB5GARcQCfQPG+GwNteIXmQAAfmQOQCgUgYBoGQDQEA2loAyLkTgpeZRmZbj3MkNQcAcYCGkpkBSG0wYuh8Sgbits8EeK8dAIJOiwnCQiVAIJ9N9F4MYExQhEBSHGPcbEjokSZKkICMQJR6JQAmMoOYmJnicnxJkiwZxLMsmVO8TJQplAmB+ICWAIJLSKBMCvFwZAAAyZAAAlYiDpER7iZggOa209ymPKf48GQSFJFNUeozReg3HhKqVEq8CkbE6SgFeJQQA) outlining this and the rationale for the `parseArgs<T extends Extact<Config, T>(config: T)` pattern proposed in [#127](https://github.com/pkgjs/parseargs/issues/127#issuecomment-1140508204). However, the comment was really mentioned to avoid casting.

I personally don't think we should promote casting in our `examples` and inlining `options` reduces the chance of immediately encountering type errors when copying `/examples/*`.

Do you think we should cast in `/examples/*`? e.g.

With JSDoc syntax
```js
const options = /** @type {const} */ ({
  foo: {
    type: 'boolean',
    short: 'f'
  },
  bar: {
    type: 'string'
  }
})

parseArgs({ options })
```

Reply:
`as const` isn't a cast; `const` isn't a type.

I think the examples should be written without TypeScript (or type comments), and there can be specific examples for using the TypeScript types which use the normal TypeScript syntax (and which use `as const`, like you would in a real program).

Reply:
@shadowspawn That will give you very imprecise types, though. Putting the options inline in the call or declaring as `as const` is the only way to get useful type inference. If we're going to be giving a TypeScript example it should use `as const`.

Reply:
> Which is the exact behavior of [const assertions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) e.g. converting wide types to literal types

No, that's not what it does. `as const` affects type _inference_: it's a hint to the type inference logic that it should choose to infer the narrowest type from the literal it's attached to, instead of guessing something more general. By contrast, a cast like `as Type` happens _after_ type inference: it takes an already-inferred type and blithely assumes it's something else (within certain constraints) without checking. They happen to share some syntax but are otherwise completely unrelated, and they interact with completely different parts of the system. (One way you can tell is that you can use `as Type` with any expression, but you are syntactically disallowed from writing `as const` for anything except a literal.)

I agree we should avoid casts, but `as const` is not a cast; there is no reason to avoid it.

> Not entirely sure what this comment is referring to

I'm pointing out that `as const` is a fundamentally different thing from `as type`.

Reply:
Thanks for the breakdown! I was purely going off definitions in the `const assertion` pull request/TS docs. This helped a ton and I'm convinced there is no reason to avoid them 🙌

Reply:
This informative sidebar means I definitely want a TS example in at least this repo! 😄  

(I have seldom hit this issue in the wild and had forgotten the details.)